prod/westeurope/functions_app1/functions_app1_r3/function_app/terragrunt.hcl

```diff
@@ -0,0 +1,205 @@
+}
+
+dependency "key_vault" {
+  config_path = "../../../common/key_vault"
+}
+
+# Include all settings from the root terragrunt.hcl file
+include {
+  path = find_in_parent_folders()
+}
+
+terraform {
+  source = "git::git@github.com:pagopa/io-infrastructure-modules-new.git//azurerm_function_app?ref=v2.1.18"
+}
+
+inputs = {
+  name                = "app1"
+  resource_group_name = dependency.resource_group.outputs.resource_name
+
+  resources_prefix = {
+    function_app     = "fn3"
+    app_service_plan = "fn3"
+    storage_account  = "fn3"
+  }
+
+  app_service_plan_info = {
+    kind     = "elastic"
+    sku_tier = "ElasticPremium"
+    sku_size = "EP1"
+  }
+
+  runtime_version = "~3"
+
+  application_insights_instrumentation_key = dependency.application_insights.outputs.instrumentation_key
+
+  app_settings = {
+    FUNCTIONS_WORKER_RUNTIME       = "node"
+    WEBSITE_NODE_DEFAULT_VERSION   = "10.14.1"
+    WEBSITE_RUN_FROM_PACKAGE       = "1"
+    FUNCTIONS_WORKER_PROCESS_COUNT = 4
+    NODE_ENV                       = "production"
+
+    COSMOSDB_URI  = dependency.cosmosdb_account.outputs.endpoint
+    COSMOSDB_KEY  = dependency.cosmosdb_account.outputs.primary_master_key
+    COSMOSDB_NAME = dependency.cosmosdb_database.outputs.name
+    // TODO: Rename to STORAGE_CONNECTION_STRING
+    QueueStorageConnection = dependency.storage_account.outputs.primary_connection_string
+    MESSAGE_CONTAINER_NAME = dependency.storage_container_message-content.outputs.name
+
+    LogsStorageConnection      = dependency.storage_account_logs.outputs.primary_connection_string
+    AssetsStorageConnection    = dependency.storage_account_assets.outputs.primary_connection_string
+    STATUS_ENDPOINT_URL        = "https://app-backend.io.italia.it/info"
+    STATUS_REFRESH_INTERVAL_MS = "300000"
+
+    // TODO: Rename to SUBSCRIPTIONSFEEDBYDAY_TABLE_NAME
+    SUBSCRIPTIONS_FEED_TABLE = dependency.storage_table_subscriptionsfeedbyday.outputs.name
+    MAIL_FROM                = "IO - l'app dei servizi pubblici <no-reply@io.italia.it>"
+    DPO_EMAIL_ADDRESS        = "dpo@pagopa.it"
+    PUBLIC_API_URL           = "http://api-internal.io.italia.it/"
+    FUNCTIONS_PUBLIC_URL     = "https://api.io.italia.it/public"
+
+    // Keepalive fields are all optionals
+    FETCH_KEEPALIVE_ENABLED             = "true"
+    FETCH_KEEPALIVE_SOCKET_ACTIVE_TTL   = "110000"
+    FETCH_KEEPALIVE_MAX_SOCKETS         = "40"
+    FETCH_KEEPALIVE_MAX_FREE_SOCKETS    = "10"
+    FETCH_KEEPALIVE_FREE_SOCKET_TIMEOUT = "30000"
+    FETCH_KEEPALIVE_TIMEOUT             = "60000"
+
+    // Push notifications
+    AZURE_NH_HUB_NAME                       = dependency.notification_hub.outputs.name
+    NOTIFICATIONS_QUEUE_NAME                = dependency.notification_queue.outputs.name
+    NOTIFICATIONS_STORAGE_CONNECTION_STRING = dependency.notification_storage_account.outputs.primary_connection_string
+
+    SLOT_TASK_HUBNAME = "ProductionTaskHub"
+
+    WEBSITE_VNET_ROUTE_ALL = 1
```

Comment:
```suggestion
```
sorry @uolter this app setting not produces any effect because function app is not using it.
@francescopersico @uolter @gunzip do we want to disable it?

PS: if we agree to enable `WEBSITE_VNET_ROUTE_ALL=1`, this app setting is missing on staging slot